src/bulk/common.ts

```diff
@@ -334,28 +335,44 @@ export class BulkWriteResult {
  * @category Error
  */
 export class WriteConcernError {
-  err: MongoServerError;
+  errCode: number;
+  errMsg: string;
+  errDetails?: Document;
+  [kErr]: { code: number; errmsg: string; errInfo?: Document };
 
-  constructor(err: MongoServerError) {
-    this.err = err;
+  constructor(error: { code: number; errmsg: string; errInfo?: Document }) {
+    this[kErr] = error;
+    this.errCode = error.code;
```

Comment:
Since this is a spec error, I don't think we should add new properties to the body (at least at this point), so I'd skip the explicit props on the class and just update the getters to read the code/errmsg/errInfo from [kErr] instead

Reply:
Ahh, okay! That's super smart. Will make that change now.